akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -53,6 +51,10 @@ private[akka] abstract class SinkModule[-In, Mat](val shape: SinkShape[In]) exte
     if ((thatN eq null) || thisN == thatN) shape
     else shape.copy(in = Inlet(thatN + ".in"))
   }
+
+  protected def label: String
+  final override def toString: String = f"$label [${System.identityHashCode(this)}%08x]"
```

Comment:
While I cannot comment on your bodily reactions I can testify that without including the object identity in the string representation it would be utterly impossible to debug materialization problems. Also, calling toString on graphs is not something people will normally do routinely, but if it they do the effect will be non-existent since we do not use the object monitors at all.